openshift/system/Dockerfile

```diff
@@ -1,91 +1,88 @@
-FROM quay.io/centos7/ruby-27-centos7 AS porta-base
-
-USER root
-
+FROM quay.io/centos/centos:stream8 as builder
+WORKDIR /opt/system
 ENV RUBY_MAJOR_VERSION=2 \
-    RUBY_MINOR_VERSION=7
+    RUBY_MINOR_VERSION=7 \
+    RAILS_ENV=production \
+    NODE_ENV=production
+ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}"
+ADD . ./
+ADD openshift/system/config/* ./config/
+RUN dnf -y module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
+    && dnf install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info make automake gcc gcc-c++ redhat-rpm-config postgresql rubygem-irb rubygem-rdoc ruby-devel nodejs libpq-devel mysql-devel gd-devel git 'dnf-command(download)' podman-catatonit \
+    && gem install bundler --version 2.2.25 \
+    && bundle config set --local without development:test:licenses \
+    && bundle config set --local deployment true \
+    && bundle install --jobs $(grep -c processor /proc/cpuinfo) --retry=5 \
+    && npm install -g yarn \
+    && yarn install:safe --no-progress \
+    && SECRET_KEY_BASE=rails/32947 bundle exec rake assets:precompile tmp:clear \
```

Comment:
@akostadinov just invented this value so it can be tracked down to https://github.com/rails/rails/issues/32947 for more information about why this is needed.
The value can be anything, in fact I think we could just have `SECRET_KEY_BASE=dummy` or something like this.

Reply:
True, although I can check whether it is still needed with Rails 6

Reply:
I checked it. They fixed it with a new env variable `SECRET_KEY_BASE_DUMMY`: https://github.com/rails/rails/pull/46760

However the patch is not included in Rails 6.0:

https://github.com/rails/rails/blob/28bb76d3efc39b2ef663dfe2346f7c2621343cd6/railties/lib/rails/application.rb#L419-L427

Besides, I don't think think there's much difference in setting `SECRET_KEY_BASE` or `SECRET_KEY_BASE_DUMMY`. The second is the official way but I don't see much improvement is this solution.

Reply:
an improvement is that `SECRET_KEY_BASE_DUMMY` is searchable and hopefully documented. While setting `SECRET_KEY_BASE` appears to be confusing. I mean you questioned that, then investigated, etc. Would have been faster with `SECRET_KEY_BASE_DUMMY`. But the fix is sub-optimal nonetheless.

Reply:
Ok, keeping this as is because it only exists in 7.0, I'll leave a todo to update later.